plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcFileWriterFactory.java

```diff
@@ -160,18 +161,20 @@ public Optional<FileWriter> createFileWriter(
         }
 
         try {
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(session.getIdentity(), path, configuration);
-            OrcDataSink orcDataSink = createOrcDataSink(fileSystem, path);
+            TrinoFileSystem trinoFileSystem = trinoFileSystemFactory.create(session);
+            OrcDataSink orcDataSink = createOrcDataSink(trinoFileSystem, path.toString());
 
             Optional<Supplier<OrcDataSource>> validationInputFactory = Optional.empty();
+            String stringPath = path.toString();
```

Comment:
if you move it higher, you can also use it in `createOrcDataSink(trinoFileSystem, path.toString());`